### PR TITLE
Use software codec for 4k videos

### PR DESCRIPTION
**Pull request by sahilbajaj**
Fixes MO-6378

### CODE REVIEW

File: lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt [hunk 1]
```diff
@@ -48,7 +46,50 @@ class Decoder(
...
+        return width * height > 2120 * 2120
+    }
+
+    private fun MediaCodecInfo.isHardwareAcceleratedCompat() : Boolean {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            isHardwareAccelerated
+        } else {
+            val isSoftware = (name.startsWith("OMX.google.") || name.startsWith("c2.android"))
+            if (isSoftware) {
+                log.d("sw codec: $name")
```
Comment (dhruv2295) [inline lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt]: there is no log.d available

File: lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt [hunk 2]
```diff
@@ -48,7 +46,50 @@ class Decoder(
     private val log = Logger("Decoder(${format.trackType},${ID[format.trackType].getAndIncrement()})")
     override val channel = this
 
-    private val codec = createDecoderByType(format.getString(MediaFormat.KEY_MIME)!!)
+    private val codec = createDecoderByType(format, format.is4K())
+
+    private fun MediaFormat.is4K(): Boolean {
+        val width = format.getInteger(MediaFormat.KEY_WIDTH)
+        val height = format.getInteger(MediaFormat.KEY_HEIGHT)
+        return width * height > 2120 * 2120
```
Comment (dhruv2295) [inline lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt:56]: lets move the numbers to companion object as const variables?

**Review by dhruv2295 — CHANGES_REQUESTED** [review]

File: lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt (hunk 1 above)
Comment (sahilbajaj) [reply lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt]: fixed

File: lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt (hunk 2 above)
Comment (sahilbajaj) [reply lib/src/main/java/com/otaliastudios/transcoder/internal/codec/Decoder.kt:56]: this should change soon. so just suppressing it until we move it

**Review by dhruv2295 — APPROVED** [review]

**Pull request merged**